### PR TITLE
feat: call start/end times

### PR DESCRIPTION
**Pull request by simonfernandes**
## Description

In the call creation dialog you can specify a time that the call will start or end. This is based on the timezone specified in a backend environment variable (e.g. 'Europe/London') and not on the timezone of the user officer creating/editing it.

To handle this, I have used Luxon time management library adapter in the Material picker. This is because `date-fns` doesn't have timezone support, and even with the `date-fns-tz` extension, it only adds some utility functions. I initially tried to get this working by implementing my own logic and overiding some `format()`, `parse()` and `label()` functions of the `date-fns` adapter, but it got very complex to implement and test. Luxon works well for this and is still being maintained unlike Moment.

Screenshots are of me based in Alaska:

## Motivation and Context

So that calls can end at a particular time instead of just the end of a day

## How Has This Been Tested

- Tested time saves correctly when creating call
- Tested time saves correctly when editing call
- Added tests for call time remaining, time field requirement
- Regarding replacing `DateFnsUtils` with `LuxonUtils`, see https://github.com/UserOfficeProject/user-office-frontend/pull/750#issuecomment-1032893690

## Fixes

UserOfficeProject/stfc-user-office-project#276

## Changes

- User officer can specify a start/end time when creating/editing a call. Timezone based on env var specified.
- In the call table, start/end call time is displayed in the env var timezone.
- All other references to the call time (e.g. the deadline when a user selects a call) are based on the user's timezone
- Async job to check call end runs hourly instead of daily
- The postgres session timezone is configured via Knex

## Depends on

- Frontend PR: https://github.com/UserOfficeProject/user-office-frontend/pull/750
- NodeJS backend timezone should be set using the `TZ` env var, with value of IANA timezone, e.g. `Europe/London`. 
  - For STFC: https://github.com/isisbusapps/docker-orchestration/pull/221

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: src/config/ess/configureESSEnvironment.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ async function enableDefaultEssFeatures() {
     ],
     true
   );
+  await db.updateSettings(SettingsId.TIMEZONE, process.env.TZ);
```
Comment (mr-field) [inline src/config/ess/configureESSEnvironment.ts:37]: Can we add some logic to default to UTC is no timezone was specified? Might be good to also log an `INFO` message if that is the case, to warn users about configuring timezones.

We should do the same in the STFC config, and in the DB init function.

File: docker-compose.e2e.yml [hunk 2]
```diff
@@ -25,6 +26,7 @@ services:
       POSTGRES_PASSWORD: duopassword
       POSTGRES_USER: duouser
       POSTGRES_DB: duo
+      PGTZ: Europe/Stockholm
```
Comment (mr-field) [inline docker-compose.e2e.yml:29]: I think we can remove this one now

**Review by mr-field — APPROVED** [review]

File: src/config/stfc/configureSTFCEnvironment.ts [hunk 3]
```diff
@@ -34,6 +35,17 @@ async function enableDefaultStfcFeatures() {
...
+  if (process.env.TZ) {
+    await db.updateSettings(SettingsId.TIMEZONE, process.env.TZ);
+  } else {
+    const defaultTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+    logger.logWarn(
+      `Timezone should be explicitly set via TZ environment variable, defaulting to: ${defaultTimezone}`,
+      {}
+    );
+    await db.updateSettings(SettingsId.TIMEZONE, defaultTimezone);
+  }
```
[the remark this answers is not in the document]
Comment (ACLay) [reply src/config/stfc/configureSTFCEnvironment.ts:48]: I do think this'd be good. Just having it in one place would mean if one facility changes how they handle timezones they can't forget to update the setup for the other

**Review by martin-trajanovski — APPROVED** [review]
Just small note for keeping the example env updated but otherwise looks great. 👀 👍

File: src/config/setTimezone.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { logger } from '@user-office-software/duo-logger';
+import { container } from 'tsyringe';
+
+import { AdminDataSource } from '../datasources/AdminDataSource';
+import { SettingsId } from '../models/Settings';
+import { Tokens } from './Tokens';
+
+/**
+ * Sets the timezone Setting from the 'TZ' environment variable,
```
[the remark this answers is not in the document]
Comment (simonfernandes) [reply src/config/setTimezone.ts:9]: That's a good suggestion, thanks, I've added it 😊

**Review by ACLay — APPROVED** [review]

**Pull request merged**